core/controllers/profile_test.py

```diff
@@ -301,8 +301,14 @@ def test_contribution_msec(self):
         # Update the first_contribution_msec to the current time in
         # milliseconds.
         first_time_in_msecs = utils.get_current_time_in_millisecs()
-        user_services.update_first_contribution_msec_if_not_set(
-            user_id, first_time_in_msecs)
+        user_settings_to_update = user_services.get_user_settings(user_id)
+        user_settings = (
```

Comment:
Can you have a method on the object instead? It would be much cleaner to replace all this with `user_settings.update_first_contribution_msec(first_time_in_msecs)`. This follows the pattern we use for e.g. exploration updates.

Ditto for other usages of user_services.update_first_contribution_msec().